### ISSUE TITLE
cli-usage.md and configuration.md reference CLI flags/env vars that clap and Config don't support

### ISSUE THREAD

**Issue by joshrotenberg** [joshrotenberg/mdbook-lint#422]
**Priority:** medium

## Current state

Cross-checking `docs/src/cli-usage.md` and `docs/src/configuration.md` against the actual `clap` `Cli`/`Commands` definition in `crates/mdbook-lint-cli/src/main.rs`:

- `docs/src/configuration.md` line 334: `mdbook-lint rules --enabled` — there is no `--enabled` flag on the `Rules` subcommand. Confirmed at runtime: `mdbook-lint rules --enabled` → `error: unexpected argument '--enabled' found`.
- `docs/src/cli-usage.md` line 67: `--format <FORMAT>` for `rules` is documented as accepting `text, JSON`, but the actual `RulesFormat` enum only has `default`/`json` (no `text` value). Confirmed at runtime: `mdbook-lint rules --format text` → `error: invalid value 'text' for '--format <FORMAT>' [possible values: default, json]`.
- `docs/src/cli-usage.md` "Basic Commands" only lists `lint`, `rules`, `rustdoc`, `help` — it omits five real subcommands that exist in `main.rs`'s `Commands` enum: `fix`, `check`, `init`, `supports`, `preprocessor` (and `lsp` when the `lsp` feature is enabled). A reader has no way to discover `mdbook-lint init`, `mdbook-lint check`, or the dedicated `mdbook-lint fix` subcommand from this page, even though `init` and `check` are referenced elsewhere in `configuration.md`.
- `docs/src/configuration.md` lines 20 and 200-201 document `MDBOOK_LINT_CONFIG` and `MDBOOK_LINT_LOG` environment variables as part of "Configuration Discovery" and "Environment Variables" respectively, and lists them in the configuration precedence order (line 195). Neither variable is read anywhere in the code (`grep -rn "MDBOOK_LINT_CONFIG\|MDBOOK_LINT_LOG" crates/` returns zero hits in source). Config discovery in `crates/mdbook-lint-cli/src/config.rs::discover_config` only walks the filesystem; there is no environment-variable override path at all.

## Desired state

- `cli-usage.md` lists all real subcommands (`preprocessor`, `lint`, `fix`, `rules`, `check`, `init`, `supports`, `rustdoc`, and `lsp` behind its feature flag), each with accurate flags pulled directly from `main.rs`.
- `configuration.md` either implements `MDBOOK_LINT_CONFIG`/`MDBOOK_LINT_LOG` (a real, common ask — explicit config path override is generally useful) or removes them from the docs and precedence list until they exist.
- `rules --format` documents only `default`/`json`; drop the `--enabled` reference or implement it as a real flag on `Rules` if it's a wanted feature (`mdbook-lint check` already validates config without listing runtime-enabled rules, so `rules --enabled` filling that gap may be worth building).

## Code example

```bash
# Documented, does not work:
mdbook-lint rules --enabled
mdbook-lint rules --format text
MDBOOK_LINT_CONFIG=./ci.toml mdbook-lint lint src/

# Works today:
mdbook-lint rules --detailed
mdbook-lint rules --format json
mdbook-lint lint --config ./ci.toml src/
```

## Implementation notes

- `crates/mdbook-lint-cli/src/main.rs`: `Commands::Rules` struct (~line 140) has no `enabled` field; `RulesFormat` enum (~line 255) has only `Default`/`Json`.
- `crates/mdbook-lint-cli/src/config.rs::discover_config` (~line 579) is the actual discovery logic — no env var consulted.
- If implementing `MDBOOK_LINT_CONFIG`, the natural place is at the top of `discover_config` or in `run_cli_mode` before the existing `config_path`/`discover_config` branch in `main.rs` (~line 906).
